main/refer.c

```diff
@@ -535,3 +537,59 @@ int ast_refer_init(void)
 	ast_register_cleanup(refer_shutdown);
 	return 0;
 }
+
+int ast_refer_notify_transfer_request(struct ast_channel *source, const char *referred_by,
+				const char *exten, const char *protocol_id,
+				struct ast_channel *dest, struct ast_refer_params *params,
+				enum ast_control_transfer state)
+{
+	RAII_VAR(struct ast_ari_transfer_message *, transfer_message, NULL, ao2_cleanup);
+	RAII_VAR(struct stasis_message *, msg, NULL, ao2_cleanup);
+	RAII_VAR(struct ast_bridge *, source_bridge, NULL, ao2_cleanup);
+	RAII_VAR(struct ast_bridge *, dest_bridge, NULL, ao2_cleanup);
+
+	transfer_message = ast_ari_transfer_message_create(source, referred_by, exten, protocol_id, dest, params, state);
+	if (!transfer_message) {
+		return -1;
+	}
+	source_bridge = ast_bridge_transfer_acquire_bridge(source);
+	if (source_bridge) {
+		RAII_VAR(struct ast_channel *, peer, NULL, ast_channel_cleanup);
+
+		ast_bridge_lock(source_bridge);
+		transfer_message->source_bridge = ast_bridge_snapshot_create(source_bridge);
+		peer = ast_bridge_peer_nolock(source_bridge, source);
+		if (peer) {
+			ast_channel_lock(peer);
+			transfer_message->source_peer = ast_channel_snapshot_get_latest(ast_channel_uniqueid(peer));
```

Comment:
This should use ast_channel_snapshot to get the snapshot and save a hash table lookup, note though that the reference count has to be incremented yourself